tests/test_piffle.py

```diff
@@ -56,3 +56,13 @@ def test_format(self):
         with pytest.raises(Exception):
             img.format('bogus')
 
+    def test_url_parse(self):
+        img = iiif.IIIFImageClient()
+        # simple API parameters
+        img.parse_from_url('http://imgserver.co/loris/img1/full/full/0/default.jpg')
+        img.api_params_as_dict()
+        # more complex API parameters
+        img.parse_from_url('http://imgserver.co/loris/img1/2560,2560,256,256/256,/!90/default.jpg')
+        img.api_params_as_dict()
```

Comment:
I would break this out into a separate unit test; probably easiest to test the submethods individually and just check that you get the expected values and types for the variety of options that are allowed for each.